Started function for general data + added comments

jasonhan3 opened the pull request:
1. Added comments with thoughts of how to parameterize app
2. Started a function that repurposes code for Dubois data to work for general data formats
3. Uncommented line 109 in app.py so that the code works
4,. Made comments to make things make more sense to me
5. Added some suggestions for how to better parameterize js code
6. Created a new requirements text file in case I need to split requirements for production vs. for locally compiling the code

jasonhan3 committed:
Started function for general data + added comments

1. Added comments with thoughts of how to parameterize app
2. Started a function that repurposes code for Dubois data to work for general data formats
3. Uncommented line 109 in app.py so that the code works
4,. Made comments to make things make more sense to me
5. Added some suggestions for how to better parameterize js code
6. Created a new requirements text file in case I need to split requirements for production vs. for locally compiling the code

u8sand commented on app/app.py:
```diff
@@ -57,6 +62,38 @@ def __call__(self, environ, start_response):
+###
+
+
+def general_plot_api(expression_file, groups_file):
```
This function doesn't do anything? Please exclude it from the PR.

u8sand commented on app/app.py:
```diff
@@ -68,7 +105,10 @@ def __call__(self, environ, start_response):
 metadata_dataframe['sample_name'] = metadata_dataframe['Sample']
-# metadata_dataframe['sample_name'] = ['sample_'+str(x).replace('.', '_') for x in metadata_dataframe['Sample']]
+# Have the same label names as logcpm columns.
+metadata_dataframe['sample_name'] = ['sample_'+str(x).replace('.', '_') for x in metadata_dataframe['Sample']]
```
Instead of doing this which is only applicable to the specific data files you're working with, I think it's better to just augment the data to be properly formatted where the metadata and data indices match.

u8sand commented on requirements.production.txt:
```diff
@@ -0,0 +1,5 @@
+Flask
+flask-basicauth
+uwsgi
```
Only `uwsgi` should be in requirements.production.txt, requirements should not be duplicated.

u8sand reviewed (changes requested):
Just some suggestions, but this is not ready to be merged.

jasonhan3 committed:
Modified code to make it more general

1. Made a general python function that should work for properly formatted data
2. Took out group strings and replaced them with group labels
3. Added comments to note to self where to tailor code for particular studies

jasonhan3 committed:
Wrote out thoughts for general Dubois code

jasonhan3 committed:
Generalized app + specified input formats

Included test data from GSE14221 for user to test as well. (Note that test data only has one sample per condition and is relatively trivial)

jasonhan3 committed:
Simplified "groups" dictionary

- Made groups dictionary map to a list of strings (rather than a list of dictionaries)

jasonhan3 committed:
Create expression/metadata files + made app general based on URL

- App now works by path ".../GSEXXXXX" for any of the 7 processed GSE studies

jasonhan3 committed:
Generalized app.py

- continuation of previous commit

jasonhan3 committed:
Displayed Metadata + Generalized Website + Added SOFT files (used by GEOparse)

- Imported GEOparse (added to requirements.txt + requirements.production.txt) and wrote geometa.py for programmatic access of metadata
- Programatically accesses and downloads metadata in SOFT files or opens SOFT files that are already downloaded for GSE's and displays metadata for each unique GSE URL
- Changed "entry_point" to "/gene-viewer"
- Made app variable based on "geo_accession" variable (.../gene-viewer/GSEXXXXX) would open a unique gene viewer for that GSE accession number, given that expression + metadata is in "data" folder under the folder name "GSEXXXXX"
- Added SOFT files to data folder (used by GEOparse module to get metadata)
- Changed style of HTML title to make it change width based on screen size

jasonhan3 committed:
Added styling + metadata to gene viewers and made home page

- Created a homepage (with a Navbar) that dynamically lists all studies in the 'data' folder
- Changed names (suppl_meta --> geo_meta; index.html --> viewer.html, yaxis_title changed from 'log10 counts per million' to axis that depends whether data is Microarray or RNA-seq
- Added programatically-accessed metadata, footer, and styled 'viewer.html' pages
- Removed test data that was used for app ('hivangroups.json', 'processedv2.txt', 'texthivanmetadata2.txt')
- Added images for 'viewer.html' footer ('dcic.png', 'mscbi_logo.png')

jasonhan3 committed:
Added homepage + navigation; optimized run speed

- Used 'cache' to reduce expensive operations; used variables to store values
- Created homepage
- Added navbar
- Added error page
- Added default gene for GSE35226-GPL6244

jasonhan3 committed:
Cleaned up text + Added 'about' page

- Cleaned up text using ftfy
- Added 'about' page
- Added 'submit issue' buttons to footer

jasonhan3 committed:
test commit; ignore

jasonhan3 committed:
Test commit (ignore)
